esprima.js

```diff
@@ -408,6 +408,13 @@
             extra.leadingComments.push(comment);
             extra.trailingComments.push(comment);
         }
+        if (extra.tokenize) {
+            comment.type = comment.type + 'Comment';
```

Comment:
Note that this is necessary since we re-use the existing comment collection. For each comment, the type is either `Line` or `Block`. However, we mix the comment with other tokens, e.g. `Numeric`, then `Line` or `Block` does not sufficiently indicating that it is a comment. Trailing it with `Comment` gives us either `LineComment` or `BlockComment`.